[pkg/ottl] Added support for timezone in Time converter (#32479)
open-telemetry/opentelemetry-collector-contrib: pkg/ottl/ottlfuncs/README.md

|`%%` | A % sign | |
|`%c` | Date and time representation | Mon Jan 02 15:04:05 2006 |

`location` specifies a default time zone canonical ID to be used for date parsing in case it is not part of `format`.

> When loading `location` collector looks for the IANA Time Zone database in the following locations in order:

Reviewer: ```suggestion
When loading `location`, this function will look for the IANA Time Zone database in the following locations in order:
```
Nit

> - the directory or uncompressed zip file named by the ZONEINFO environment variable,

Reviewer 2: ```suggestion
- a directory or uncompressed zip file named by the ZONEINFO environment variable
```

> - on a Unix system, the system standard installation location,

Reviewer 2: ```suggestion
- on a Unix system, the system standard installation location
```

> - $GOROOT/lib/time/zoneinfo.zip,

Reviewer 2: ```suggestion
- $GOROOT/lib/time/zoneinfo.zip
```

- the `time/tzdata` package, if it was imported. 

> In case you're building from source importing `time/tzdata` will ensure collector works properly on each system. This will add roughly 500kB to binary size.

Reviewer: ```suggestion
When building a Collector binary, importing `time/tzdata` in any Go source file will bundle the database into the binary, which guarantees the lookups will work regardless of the setup on the host setup. Note this will add roughly 500kB to binary size.
```
I think we should just add a bit more detail on what this process could look like.
